typed-racket-lib/typed-racket/infer/infer-unit.rkt

```diff
@@ -513,29 +517,31 @@
 
         ;; recursive names should get resolved as they're seen
         [(s (? Name? t))
-         (cg s (resolve-once t))]
+         (let ([t (resolve-once t)])
```

Comment:
no -- `obj` can be false and that shouldn't collapse the computation (i.e. it's not an indication of 'failure'). That's the other reason I don't like `%`, it's really easy to get so used to using it you can w/o thinking introduce a bug into your code